core/iwasm/common/wasm_shared_memory.c

```diff
@@ -322,6 +374,9 @@ wasm_runtime_atomic_wait(WASMModuleInstanceCommon *module, void *address,
     bh_assert(module->module_type == Wasm_Module_Bytecode
               || module->module_type == Wasm_Module_AoT);
 
+    if (wasm_get_exception(module_inst)) {
+        return -1;
+    }
```

Comment:
Had better add an empty line below.

Reply:
And found this PR is for `dev/wasi_threads`, could you submit to main first? We will merge main into dev/wasi_threads after several PRs are merged.

Reply:
i think changing base to main will cause all commits to dev/wasi_threads added to this PR (since the branch was created from dev/wasi_threads), will create a new PR

Reply:
You can keep this PR, but change base of your `stale_threads_term` branch to be `main` instead of `dev/wasi_threads`.

Reply:
done